### PR TITLE
Add homepage illustration (and other tweaks)

### LINKED ISSUE #9: Homepage illustration
@paulrobertlloyd has volunteered. 😁

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
Adds homepage illustration, Open Graph share image, and reviews content such that:

* code examples are correctly indented
* pages are listed in the sub navigation in the same order as the homepage
* product name is capitalised (to match other X-GOVUK products)
* correct heading styles are used on the homepage

Also added a `.editorconfig` file. Closes #9. 💅

### CODE REVIEW

File: docs/assets/opengraph-image.png
Comment (frankieroberto) [inline docs/assets/opengraph-image.png:1]: The title in here should be plural, no?

File: docs/index.njk [hunk 1]
```diff
@@ -2,27 +2,30 @@
...
 <div class="govuk-grid-row">
   <section class="govuk-grid-column-two-thirds-from-desktop">
-  {{ content | markdown }}
+    <h2 class="govuk-heading-m govuk-!-font-size-27">Navigating and filling in forms</h2>
+    <ul class="govuk-list govuk-list--bullet">
+      <li><a href="click-govuk-button">Buttons</a></li>
+      <li><a href="within-fieldsets">Fieldsets</a></li>
+      <li><a href="click-govuk-link">Links</a></li>
+      <li><a href="choosing-radio-options">Radio options</a></li>
+      <li><a href="fill_in_govuk_text_field">Text fields</a></li>
```
Comment (frankieroberto) [inline docs/index.njk:21]: This should have the govuk-link class? (Maybe the docs should have some rspec tests which use the helpers they describe, inception-style... 😂)
```suggestion
      <li><a class="govuk-link" href="click-govuk-button">Buttons</a></li>
      <li><a class="govuk-link" href="within-fieldsets">Fieldsets</a></li>
      <li><a class="govuk-link" href="click-govuk-link">Links</a></li>
      <li><a class="govuk-link" href="choosing-radio-options">Radio options</a></li>
      <li><a class="govuk-link" href="fill_in_govuk_text_field">Text fields</a></li>
```

File: docs/index.njk [hunk 2]
```diff
@@ -2,27 +2,30 @@
...
+    </ul>
+    <h2 class="govuk-heading-m govuk-!-font-size-27">Expecting content</h2>
+    <ul class="govuk-list govuk-list--bullet">
+      <li><a href="summarise-errors">Error summaries</a></li>
+      <li><a href="summarise">Summary lists</a></li>
```
Comment (frankieroberto) [inline docs/index.njk:26]:
```suggestion
      <li><a class="govuk-link" href="summarise-errors">Error summaries</a></li>
      <li><a class="govuk-link" href="summarise">Summary lists</a></li>
```

File: docs/index.njk [hunk 3]
```diff
@@ -2,27 +2,30 @@
...
+    </ul>
+    {{ content | markdown | replace("govuk-heading-l", "govuk-heading-m govuk-!-font-size-27") }}
```
Comment (frankieroberto) [inline docs/index.njk:28]: Is this line needed if all the content is in HTML above now?

**Review by frankieroberto — CHANGES_REQUESTED** [review]
Thanks for the great illustration and all the small fixes! 🙌 

Left a couple of minor comments.

File: docs/index.njk (hunk 3 above)
Comment (paulrobertlloyd) [reply docs/index.njk]: Oh crap, this was what I tried to begin with, which, while it worked, seemed a bit over the top and possibly unnecessary given the page already contains handcrafted artisanal HTML.

File: docs/assets/opengraph-image.png
Comment (paulrobertlloyd) [reply docs/assets/opengraph-image.png:1]: Ah, good spot. Fixed.

**Review by frankieroberto — APPROVED** [review]

**Pull request merged**